modules/index/src/main/java/org/apache/ignite/internal/index/IndexBuildController.java

```diff
@@ -554,10 +563,23 @@ private static IndexStorage indexStorage(
             int partitionId,
             CatalogIndexDescriptor indexDescriptor
     ) {
-        IndexStorage indexStorage = mvTableStorage.getIndex(partitionId, indexDescriptor.id());
+        IndexStorage indexStorage;
+        try {
+            indexStorage = mvTableStorage.getIndex(partitionId, indexDescriptor.id());
+        } catch (StorageClosedException e) {
+            throw new TableClosedException(indexDescriptor.tableId(), e);
+        }
 
         assert indexStorage != null : "Index storage is missing [partitionId=" + partitionId + ", indexId=" + indexDescriptor.id() + "].";
 
         return indexStorage;
     }
+
+    private static class TableClosedException extends IgniteInternalException {
```

Comment:
I prefer the practice of using a packaged private class instead of an internal one, I would change it, at your discretion.